### PR TITLE
Update note and tracking fields workflow

### LINKED ISSUE #9397: Notes on orders: Immediately edit or delete the note (and tracking information)
## Description

Follow up of #9368 
Some improvements of usability and to make sure the implementation is according to specification ([see mockup here](https://github.com/openfoodfoundation/openfoodnetwork/issues/9178#issue-1234182060)).

## Expected Behavior

- As soon as clicking the 'edit' icon, the cursor should be placed at the end of the note to start typing without an extra click.
- Add the 'delete' icon to immediately delete the note without having to edit and delete the text. When hovering over the delete icon, use the red styling and the crossed out text styling as on line items.
- Both should be implemented for the tracking information at the same time.

## Actual Behaviour

- After clicking the 'edit' icon you have to click inside the text field first to start typing.
- There is no direct 'delete' icon.
- Same for tracking information.

## Steps to Reproduce

1. Go to /admin/orders/[order-number]/edit.
2. Try to edit or delete the note.

## Animated Gif/Screenshot

grafik

## Severity
bug-s4: it's annoying, but you can use it

## Your Environment

* Version used: staged [89809aed5fdcd953d13b9108290c942a5182ceb4](https://github.com/openfoodfoundation/openfoodnetwork/pull/9368/commits/89809aed5fdcd953d13b9108290c942a5182ceb4)
* Browser name and version: Firefox 102.0 (64-Bit)
* Operating System and version (desktop or mobile): Desktop Windows 10

### PR DESCRIPTION
**Pull request by binarygit**
#### What? Why?
Closes #9397 

#### What should we test?

- Visit Page /admin/orders/#{order_number}/edit
- Create a note/tracking info
- Delete the note/tracking info using the new delete button
- Check if the note/tracking info is deleted

#### Release notes

Changelog Category: User facing changes

The title of the pull request will be included in the release notes.

### CODE REVIEW

**Comment by binarygit** [comment]
@drummer83 Hey :smiley: 
I had some time today so I went ahead and did this. Let me know if any changes are required.
SS:
ezgif com-gif-maker(9)

**Comment by drummer83** [comment]
Hi @binarygit,
Great to see you working on this! 💪 

At first sight this looks great to me. 👍 Your pull request is now in Code Review and our developers will let you know if changes are required, it will be tested and then released.

I will create a follow up issue to implement the red highlighting and the crossed out text styling when hovering over the delete icon. This request was a little it hidden in the issue:

> When hovering over the delete icon, use the red styling and the crossed out text styling as on line items.

If you feel like adding this to your current pull request please comment here so I can move back to In Dev. Otherwise let's wait until this one is merged and then you or someone else can pick up the last bits.

Thanks again and let me know what you think!

**Comment by drummer83** [comment]
Oh, can you please comment on the original issue #9397 so I can assign it to you, @binarygit!

**Comment by binarygit** [comment]
@drummer83 Why don't I pick up the other requirement after this one gets merged? I am working on another PR too, and I want to finish that first.

**Comment by jibees** [comment]
Hey @binarygit 
Thanks a lot for this one. Could you please fill in the description of the PR as indicated in the template?

**Comment by binarygit** [comment]
@jibees Sorry, I must've been in a hurry when I opened this! :sweat_smile: 
I think I filled it out correctly this time...let me know if I missed something!

File: app/views/spree/admin/orders/_note.html.haml
Comment (jibees) [inline app/views/spree/admin/orders/_note.html.haml:6]: Is it voluntary that you've put the `text_field` into the `label` tag? I'm wondering why you did this, even if it seems relevant in certains cases for sure (https://www.w3.org/WAI/tutorials/forms/labels/#associating-labels-implicitly)

**Review by jibees — CHANGES_REQUESTED** [review]
Thanks a lot for this one!
Just small comments, but more generally I think this two changes (for tracking and note) deserves a spec for each one.

Probably into the spec file `spec/system/admin/order_spec.rb` (see commit 576faff3d7dc3ed6eeca389b7695be5c2eea1cf2 for example)

File: app/views/spree/admin/orders/_note.html.haml
Comment (binarygit) [reply app/views/spree/admin/orders/_note.html.haml:6]: Honestly, I thought you'd meant to put the `text_field` into the `label` tag but somehow it got screwed up. That's why I put it inside the label tag. I just thought it was good practice to have labels associated with input elements, no? :shrug:

**Comment by drummer83** [comment]
Hi @binarygit,
Thank you for this amazing pull request! I have tested and can confirm that usability has very much improved! Really cool! 💪 

The following situations have been tested:

As an enterprise user on a confirmed order, cancelled order, resumed order:
- entering a note --> ok
- editing a note --> ok
- deleting a note --> ok
- entering tracking information --> ok
- editing tracking information --> ok
- deleting tracking information --> ok
- shipping email contains up to date tracking information --> ok

As a super admin user confirmed order, cancelled order, resumed order
- entering a note --> ok
- editing a note --> ok
- deleting a note --> ok
- entering tracking information --> ok
- editing tracking information --> ok
- deleting tracking information --> ok
- shipping email contains up to date tracking information --> ok

Notes (to be handled in separate issues/pull requests):
- Display 'delete' icon only when note or tracking information is present.
image
- Add a confirmation dialog: Are you sure? when deleting a note or tracking information.
image
- If there was a tracking information added, which then has been deleted, in the shipping email there will be the 'Tracking information:' row without any further information
image

This pull request is great improvement!
It is ready to go! 🥳 

Thank you again!

File: app/views/spree/admin/orders/_note.html.haml
Comment (mkllnk) [reply app/views/spree/admin/orders/_note.html.haml:6]: Great. This is useful to associate the textfield with the label without using ids. I think it's good practice to always have either: put the input inside the label or use an id to associate the two.

**Pull request merged**
